frontend/src/composables/TabPanels/useChildsTabData.js

```diff
@@ -0,0 +1,141 @@
+import { useRouter } from "vue-router";
+import { useQuasar } from "quasar";
+
+export default function useChildsTabData(props) {
+	const store = useStore();
+	const route = useRouter();
+	const quasar = useQuasar();
+	const childId = ref(props.id);
+	const workspaceChilds = ref(null);
+	const childName = ref(props.name);
+	const childLogo = ref(props.logo);
+	const childShortDescription = ref(props.shortDescription);
+	const childDescription = ref(props.description);
+	const childRepositoryURL = ref(props.repositoryURL);
+	const childParentID = ref(route.currentRoute.value.params.id);
+
+	const getChilds = async () => {
+		await store.dispatch("appDomain/fetchDomainesTree");
+		let data = store.getters["appDomain/allDomainesTree"];
+		workspaceChilds.value = [...data.Childs];
+	};
+	getChilds();
+
+	const refreshChilds = async (id, updatesData) => {
+		await store.dispatch("appDomain/fetchDomainById", id);
+		await store.getters["appDomain/fetchAllDomaines"];
+		childName.value = updatesData.Name;
+		childLogo.value = updatesData.Logo;
+		childShortDescription.value = updatesData.ShortDescription;
+		childRepositoryURL.value = updatesData.GitURL;
+		childDescription.value = updatesData.Description;
+		childParentID.value = updatesData.DomainID;
+	};
+
+	const confirmDeleteChild = (id) => {
+		quasar
+			.dialog({
+				title: "Confirm",
+				message: "Are you sure to delete this item?",
+				cancel: true,
+				persistent: true,
+			})
+			.onOk(() => {
+				deleteChild(id);
+			});
+	};
+
+	const deleteChild = async (child) => {
+		try {
+			await store.dispatch("appDomain/removeDomain", `${child}`);
+			await getChilds();
+			quasar.notify({
+				type: "positive",
+				message: `Workspace was succefully deleted`,
+			});
+		} catch (error) {
+			quasar.notify({
+				type: "negative",
+				message: `Error while deleting workspace`,
+			});
+		}
+	};
+	const addNewChild = async () => {
+		let newChild = {
+			pid: route.currentRoute.value.params.id,
+			Name: childName.value,
+			Logo: childLogo.value,
+			ShortDescription: childShortDescription.value,
+			Description: childDescription.value,
+		};
+		try {
+			await store
+				.dispatch("appDomain/addDomain", newChild)
+				.then(() => {
+					getChilds();
+				})
+				.then(() => {
+					quasar.notify({
+						type: "positive",
+						message: `${childName.value} child was successfully created`,
+					});
+				});
+		} catch (error) {
+			quasar.notify({
+				type: "negative",
+				message: `${childName.value} child was not created`,
+			});
+		}
+		childName.value = "";
+		childLogo.value = "";
+		childShortDescription.value = "";
+		childDescription.value = "";
+	};
+
+	const updateChild = async () => {
+		let updates = {
+			ID: props.id,
+			ParentID: childParentID.value,
+			Name: childName.value,
+			Logo: childLogo.value,
+			ShortDescription: childShortDescription.value,
+			GitURL: childRepositoryURL.value,
+			Description: childDescription.value,
+		};
+		try {
```

Comment:
don't mix try catch and promise. in this case use only promise

Reply:
ok